sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -262,7 +262,12 @@ case class HashPartitioning(expressions: Seq[Expression], numPartitions: Int)
    * Returns an expression that will produce a valid partition ID(i.e. non-negative and is less
    * than numPartitions) based on hashing expressions.
    */
-  def partitionIdExpression: Expression = Pmod(new Murmur3Hash(expressions), Literal(numPartitions))
+  def partitionIdExpression(useHiveHash: Boolean = false): Expression =
+    if (useHiveHash) {
+      Pmod(new HiveHash(expressions), Literal(numPartitions))
```

Comment:
@viirya Thanks a lot for comment !
I compared code between v0.13 and v1.2.1. In my understanding, there's no compatibility issue. v1.2.1 add `hashcode` support for more types(`INTERVAL_YEAR_MONTH`, `INTERVAL_DAY_TIME`), for the existing types, they are compatible with each other.